cmrf_views/src/Plugin/views/query/API4.php

```diff
@@ -0,0 +1,525 @@
+
+    return $alias;
+  }
+
+  /**
+   * Returns the alias for the given field added to $table.
+   *
+   * @access protected
+   *
+   * @see \Drupal\cmrf_views\Plugin\views\query\API::addField()
+   */
+  protected function getFieldAlias($table_alias, $field) {
+    $field = str_replace('.', '__', $field);
+    return isset($this->fieldAliases[$table_alias][$field]) ? $this->fieldAliases[$table_alias][$field] : FALSE;
+  }
+
+  /**
+   * Executes the query and fills the associated view object with according
+   * values.
+   *
+   * Values to set: $view->result, $view->total_rows, $view->execute_time,
+   * $view->current_page.
+   */
+  public function execute(ViewExecutable $view) {
+    $table_data = $this->viewsData->get($view->storage->get('base_table'));
+    if (!empty($table_data)) {
+      $api_entity       = $table_data['table']['base']['entity'];
+      $api_action       = $table_data['table']['base']['action'];
+      $api_version      = $table_data['table']['base']['api_version'];
+      $connector        = $table_data['table']['base']['connector'];
+      $dataset_params   = $table_data['table']['base']['params'] ?? [];
+      if (!is_array($dataset_params)) {
+        $dataset_params = [];
+      }
+
+      $parameters = [];
+      $start      = microtime(TRUE);
+
+      // Set the return fields
+      $parameters['select'] = [];
+      foreach ($view->field as $field) {
+        if (!empty($table_data[$field->field]['cmrf_original_definition']['name'])) {
+          $original_field_name = $table_data[$field->field]['cmrf_original_definition']['name'];
+          if (!in_array($original_field_name, $parameters['return'])) {
+            $parameters['select'][] = $original_field_name;
+          }
+        }
+      }
+
+      // Set the query parameters.
+      if (!empty($this->where)) {
+        foreach ($this->where as $where_group) {
+          if (empty($where_group['conditions'])) {
+            continue;
+          }
+          foreach ($where_group['conditions'] as $condition) {
+            $original_field_name = $table_data[$condition['field']]['cmrf_original_definition']['name'];
+            switch ($condition['operator']) {
+              case '>':
+              case '>=':
+              case '<=':
+              case '<':
+              case '!=':
+              case 'BETWEEN':
+              case 'NOT BETWEEN':
+              case 'LIKE':
+              case 'NOT LIKE':
+                $parameters['where'][] = [$original_field_name, $condition['operator'], $condition['value']];
+                break;
+              case 'in':
+                $parameters['where'][] = [$original_field_name, 'IN', $condition['value']];
+                break;
+              case 'not in':
+                $parameters['where'][] = [$original_field_name, 'NOT IN', $condition['value']];
+                break;
+              default:
+                $parameters['where'][] = [$original_field_name, '=', $condition['value']];
```

Comment:
What it make sense to assign `$condition['operator'] ?? '='` to `$operator`, `switch` over `$operator`, add `case '='`, and then to throw an exception in the default case?

Reply:
I thought about that as well, but we're implementing all operators used by the CiviCRM API and it has been implemented like that ever since. Feel free to change that if you find the time.